### PR TITLE
feat: support background DDL for MV on shared source

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
follow up of #18112, where we only implemented the track mechanism in meta, but it's banned in frontend

https://github.com/risingwavelabs/risingwave/issues/18338

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18919?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#18919** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18919?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#18749** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18749?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#18894** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18894?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#18891** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18891?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#18890** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18890?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @xxchan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by xxchan** [comment]
kindly ping reviewers here

**Comment by kwannoel** [comment]
Will have a look by today

File: src/frontend/src/optimizer/plan_node/utils.rs [hunk 1]
```diff
@@ -377,17 +377,21 @@ pub fn infer_kv_log_store_table_catalog_inner(
 
 /// Check that all leaf nodes must be stream table scan,
 /// since that plan node maps to `backfill` executor, which supports recovery.
-pub(crate) fn plan_has_backfill_leaf_nodes(plan: &PlanRef) -> bool {
+/// Some other leaf nodes like `StreamValues` do not support recovery, and they
+/// cannot use background ddl.
```
Comment (BugenZhao) [inline src/frontend/src/optimizer/plan_node/utils.rs:381]: Do we have an issue for this? Is it possible to address this by simply recording the state into a state table?

**Review by BugenZhao — APPROVED** [review]

File: src/frontend/src/optimizer/plan_node/utils.rs [hunk 2]
```diff
@@ -377,17 +377,21 @@ pub fn infer_kv_log_store_table_catalog_inner(
...
+pub(crate) fn plan_can_use_backgronud_ddl(plan: &PlanRef) -> bool {
```
Comment (kwannoel) [inline src/frontend/src/optimizer/plan_node/utils.rs:382]: typo `background`

File: src/frontend/src/optimizer/plan_node/utils.rs [hunk 3]
```diff
@@ -377,17 +377,21 @@ pub fn infer_kv_log_store_table_catalog_inner(
...
     if plan.inputs().is_empty() {
-        if let Some(scan) = plan.as_stream_table_scan() {
+        if plan.as_stream_source_scan().is_some() {
```
Comment (kwannoel) [inline src/frontend/src/optimizer/plan_node/utils.rs:384]: For `stream_source_scan` it is only used by source backfill right? If so this LGTM.

**Pull request merged**